content/post/2019-12-19-mutable-api.markdown

```diff
@@ -321,6 +348,8 @@ desc::desc_add_author_gh("<githubhandle>")
 
 And the local DESCRIPTION file will be updated. So what's become mutable is the DESCRIPTION file itself via an object that's written to disk each time it's changed!
 
+To mention another `R6` example and to come back to system processes: `processx` uses R6 + Xptr, because some of the mutable state is in R. `ps` uses Xptr and then all the mutable state is in C (e.g. `ps::ps_suspend()`) or is external (what happens to the process outside of R).
```

Comment:
Xptr -> an external pointer, 

> (e.g. `ps::ps_suspend()`) or is external (what happens to the process outside of R).

I am not sure what you mean by this.

Reply:
I wanted to make explicit what the mutable part "in C" and the "external" mutable part were, better not to?

Reply:
IDK what this means TBH. `ps::ps_suspend()` is a method, not a mutable part. Yes, what we know about the process is in C, and the state of the process itself can change externally, i.e. the process might just end.